File: android/model/src/main/kotlin/com/dgsd/hackernews/model/Comment.kt
package com.dgsd.hackernews.model

public data class Comment(
        val id: Long = -1,
        val time: Long = -1,
        val author: String? = null,
        val text: String? = null,
        var parentId: Long = -1,
        val commentCount: Int = -1,
        val commentIds: List<Long> = emptyList(),
        val comments: List<Comment> = emptyList(),
        val deleted: Boolean = false,
        val dead: Boolean = false) {

    fun hasBeenRemoved(): Boolean {
        return deadOrDeleted() && commentIds.isEmpty()
    }

    fun deadOrDeleted(): Boolean {
        return deleted || dead
    }

}

File: ios/hackernews/models/Comment.swift
//
//  Comment.swift
//  hackernews
//
//  Created by Daniel Grech on 12/10/2015.
//  Copyright © 2015 DGSD. All rights reserved.
//

import Foundation

struct Comment {
    var id: Int64 = -1
    var time: Int64 = -1
    var author: String? = nil
    var text: String? = nil
    var parentId: Int64 = -1
    var commentCount: Int32
    var commentIds: [Int64] = []
    var comments: [Comment] = []
    var deleted: Bool
    var dead: Bool
    
    func hasBeenRemoved() -> Bool {
        return deadOrDeleted() && commentIds.isEmpty
    }
    
    func deadOrDeleted() -> Bool {
        return dead || deleted
    }
    
}